### DISCUSSION TITLE
Possible to automate builds, and use "Releases" in GitHub to distribute compiled version?

### DISCUSSION THREAD

**Discussion by austinwitherspoon** [AcademySoftwareFoundation/OpenRV#82] [Ideas]
Speaking as somebody who is a python developer primarily, and who has struggled trying to bring other tools like this into our pipeline, the compilation requirements/steps for c++ apps are pretty painful, and I imagine I'm not the only person to spend days trying to get everything set up on my machine to compile tools like these.

Forgive me if this is too difficult of an ask, I speak from complete ignorance of how most of this works, but it seems like it would be incredibly useful to leverage github actions to generate compiled binaries for Windows, MacOS, and Linux to more easily distribute.

Is this something at all possible?

**Comment by geffrak** [comment]
Hi @austinwitherspoon, if everything goes as planned, we will eventually have a multi-platform CI. Then, we will be able to define if we can/should publish pre-compiled builds.

**Reply by villasenor** [reply]
Hello! Just checking in - what is the status of this initiative in 2026? Is there any chance that pre-compiled builds will land soon? Trying to decide if it is worth building a CI pipeline myself to do these builds. Thanks!

**Reply by michimussato** [reply]
@villasenor three and a half years. Although @geffrak stated that OpenRV is owned by ASWF we can still see Autodesk involvement based on the maintainers' GitHub usernames. That said, I get a strong feeling that Autodesk is therefore involved in strategic decisions made in ASWF projects (possibly in favor of their own review player). 

I started a personal project a while ago, and one component of that is an automated OpenRV build pipeline - it's not public yet but once I get successful enough results with it, it'll be open sourced. Here are some announcements in case you're interested:
- https://www.linkedin.com/feed/update/urn:li:activity:7477788672021430272
- https://www.linkedin.com/feed/update/urn:li:activity:7480913001890537472
- https://www.linkedin.com/feed/update/urn:li:activity:7481950499521499137

For now, it's Linux only (tarball and Apptainer).
In parallel, I'll be working on a similar pipeline for Dnegs xSTUDIO so that people can choose more easily.

**Reply by villasenor** [reply]
Thanks for the info @michimussato! Excited to see your OpenRV build pipeline when it is ready :)

**Reply by helge000** [reply]
@michimussato, thanks - this is exactly my impression as well. I started with a private fork long time ago to get the CI run our internal gitlab instance; but did not get far (mostly because of the absence of upstream release tags). 
If you publish it would be highly appreciated by the community. Maybe let's fork this as well. 
Cheers!

**Reply by helge000** [reply]
@villasenor, thanks for reminding all of us (well, me at least)!

**Reply by michimussato** [reply]
@helge000 @villasenor The OpenRV-Builder is almost ready. Should be a matter of just a few days. Cheers!

**Reply by jarak-lvly** [reply]
Hi @michimussato,

I put together a proof-of-concept for building OpenRV CY2025 on Windows from a Linux host, using dockur/windows (QEMU inside a Docker container) rather than a dedicated Windows box: https://github.com/jarak-lvly/openrv-windows-docker-build-cy2025

It's docs, Docker Compose, and setup scripts, not a redistribution of binaries. It's intended to provide a repeatable build environment rather than build automation. Switching between Linux/Windows/macOS gets cumbersome, so that's what pushed me toward the Docker PoC.

I thought it might be a useful reference for your OpenRV-Builder.

Your OpenRV-Builder sounds cool; looking forward to trying it out!

**Reply by michimussato** [reply]
@jarak-lvly @villasenor @helge000 

[OpenStudioLandscapes-OpenRV-Builder](https://github.com/michimussato/OpenStudioLandscapes-OpenRV-Builder) is now publicly available! Don't hesitate to share your thoughts and experiences!

[![ Logo OpenStudioLandscapes ](https://github.com/michimussato/OpenStudioLandscapes/raw/main/media/images/logo128.png)](https://github.com/michimussato/OpenStudioLandscapes-OpenRV-Builder) 💙 [![Open RV](https://github.com/AcademySoftwareFoundation/OpenRV/raw/main/docs/images/OpenRV_icon.png)](https://github.com/AcademySoftwareFoundation/OpenRV.git)

- [LinkedIn Post](https://www.linkedin.com/feed/update/urn:li:activity:7488163787519295488)

**Comment by helge000** [comment]
@geffrak, may I ask about this? For some reason [Autodesk states](https://www.autodesk.com/support/technical/article/caas/sfdcarticles/sfdcarticles/RV-Open-Source-Frequently-Asked-Questions.html#:~:text=Downloadable%20executable) that binaries and downloads are not a feature of OpenRV.
That said, 
- will you accept pull requests for github CI?
- When can we expect release tags?

Thanks!

**Comment by geffrak** [comment]
@helge000, it's something that need to be defined at the ASWF level. OpenRV is owned by AWSF's open review initiative. 

It might be confusing because it seems to be the same team between both OpenRV and Autodesk ShotGrid RV, but it's two different entity with two different requirements and commercial authorization/agreement. 

It's in the ASWF roadmap to move forward about pre-compiled something to ease up the adoption.

**Reply by helge000** [reply]
@geffrak , thanks for the clarification! Oh well, I guess that means you are *not* currently accepting PR's for Github CI.

**Reply by geffrak** [reply]
You are correct. It would stay open for a moment. Until we have the go from ASWF.

**Comment by michimussato** [comment]
Hello! What is the progress on this?
